assets/l10n/app_en.arb

```diff
@@ -27,18 +31,298 @@
-  "@cameraAccessDeniedMessage": {
-    "description": "Message for dialog when the user needs to grant permissions for camera access."
+  "permissionsNeededTitle": "Permissions needed",
+  "@permissionsNeededTitle": {
+    "description": "Title for dialog when the user needs to grant additional permissions."
   },
-  "cameraAccessDeniedButtonText": "Open settings",
-  "@cameraAccessDeniedButtonText": {
+  "permissionsNeededOpenSettings": "Open settings",
+  "@permissionsNeededOpenSettings": {
+    "description": "Button label for permissions dialog button that opens the system settings screen."
+  },
+  "permissionsNeededCameraAccessDenied": "To upload an image, please grant Zulip additional permissions in Settings.",
+  "@permissionsNeededCameraAccessDenied": {
     "description": "Message for dialog when the user needs to grant permissions for camera access."
   },
+  "permissionsReadExternalStorageDenied": "To upload files, please grant Zulip additional permissions in Settings.",
+  "@permissionsReadExternalStorageDenied": {
+    "description": "Message for dialog when the user needs to grant permissions for external storage read access."
+  },
+  "actionSheetOptionCopy": "Copy message text",
+  "@actionSheetOptionCopy":  {
+    "description": "Label for copy message text button on action sheet."
+  },
+  "actionSheetOptionShare": "Share",
+  "@actionSheetOptionShare":  {
+    "description": "Label for share button on action sheet."
+  },
+  "actionSheetOptionQuoteAndReply": "Quote and reply",
+  "@actionSheetOptionQuoteAndReply":  {
+    "description": "Label for Quote and reply button on action sheet."
+  },
+  "errorCouldNotFetchMessageSource": "Could not fetch message source",
+  "@errorCouldNotFetchMessageSource": {
+    "description": "Dialog message when the source of a message could not be fetched."
+  },
+  "errorCopyingFailed": "Copying failed",
+  "@errorCopyingFailed": {
+    "description": "Dialog message when copying the text of a message to the users system clipboard failed."
+  },
+  "errorFailedToUploadFile": "Failed to upload file: {filename}",
+  "@errorFailedToUploadFile": {
+    "description": "Label in compose box showing the specified file failed to upload.",
+    "placeholders": {
+      "filename": {
+        "type": "String",
+        "example": "file.txt"
+      }
+    }
+  },
+  "errorFilesTooLarge": "{num, plural, =1{File is} other{{num} files are}} larger than the server's limit of {maxFileUploadSizeMib} MiB and will not be uploaded:\n\n{listMessage}",
+  "@errorFilesTooLarge": {
+    "description": "Error message when attached files are too large in size.",
+    "placeholders": {
+      "num": {
+        "type": "int",
+        "example": "4"
+      },
+      "maxFileUploadSizeMib": {
+        "type": "int",
+        "example": "15"
+      },
+      "listMessage": {
+        "type": "String",
+        "example": "foo.txt\nbar.txt"
```

Comment:
nit: make the example data consistent — if it's two filenames, then the number should be 2